src/components/settings/ImportAllDialog/__tests__/useGlobalImportFileParser.test.ts

```diff
@@ -0,0 +1,100 @@
+      },
+    })
+    const { result } = renderHook(() => useGlobalImportJsonParser(jsonData))
+
+    const { addedSafes, addedSafesCount, addressBook, addressBookEntriesCount } = result.current
+
+    // No addressbook data
+    expect(addressBookEntriesCount).toEqual(0)
+    expect(addressBook).toEqual(undefined)
+
+    expect(addedSafesCount).toEqual(2)
+    expect(addedSafes).toBeDefined()
+    if (!addedSafes) {
+      fail('No added safes found')
+    }
+    expect(addedSafes['5'][goerliSafeAddress]).toBeDefined()
+    const goerliAddedSafe = addedSafes['5'][goerliSafeAddress]
+    expect(goerliAddedSafe.threshold).toEqual(2)
+
+    expect(addedSafes['1'][mainnetSafeAddress]).toBeDefined()
+    const mainnetAddedSafe = addedSafes['1'][mainnetSafeAddress]
+    expect(mainnetAddedSafe.threshold).toEqual(1)
+  })
+
+  it('should parse address book entries correctly', () => {
+    const goerliAddress1 = '0xAecDFD3A19f777F0c03e6bf99AAfB59937d6467b'
+    const goerliName1 = 'test.eth'
+    const goerliAddress2 = '0x3819b800c67Be64029C1393c8b2e0d0d627dADE2'
+    const goerliName2 = 'some.eth'
+    const mainnetAddress1 = '0x954cD69f0E902439f99156e3eeDA080752c08401'
+    const mainnetName1 = 'mobile owner'
+    const mainnetAddress2 = '0x7cB6E6Cbc845e79d9CA05F6577141DA36ad398f5'
+    const mainnetName2 = 'S0mE&W3!rd#N4m€'
+
+    const jsonData = JSON.stringify({
+      version: '1.0',
+      data: {
+        SAFE__addressBook: `[{"address":"${mainnetAddress1}","name":"${mainnetName1}","chainId":"1"},
+      {"address":"${mainnetAddress2}","name":"${mainnetName2}","chainId":"1"},
+      {"address":"${goerliAddress1}","name":"${goerliName1}","chainId":"5"},
+      {"address":"${goerliAddress2}","name":"${goerliName2}","chainId":"5"}]`,
+      },
+    })
+
+    const { result } = renderHook(() => useGlobalImportJsonParser(jsonData))
+
+    const { addedSafes, addedSafesCount, addressBook, addressBookEntriesCount } = result.current
+
+    // no added safes
+    // No addressbook data
+    expect(addedSafesCount).toEqual(0)
+    expect(addedSafes).toEqual(undefined)
+
+    expect(addressBookEntriesCount).toEqual(4)
+    if (!addressBook) {
+      fail('No addressbook migrated')
+    }
```

Comment:
The assertion makes this fail redundant.

Reply:
I only added it to have `addressBook` better typed in the following lines. I'll remove the previous assertion.